src/react-imgix-bg.jsx

```diff
@@ -1,120 +1,155 @@
 import React from "react";
 import { withContentRect } from "react-measure";
-import { PACKAGE_VERSION } from './constants';
+import { PACKAGE_VERSION } from "./constants";
 import constructUrl from "./constructUrl";
 import extractQueryParams from "./extractQueryParams";
 import findClosest from "./findClosest";
 import targetWidths from "./targetWidths";
 
-
 const noop = () => {};
 
 const findNearestWidth = (actualWidth) =>
   findClosest(actualWidth, targetWidths);
 
 const toFixed = (dp, value) => +value.toFixed(dp);
 
-const BackgroundImpl = (props) => {
-  const {
-    measureRef,
-    measure,
-    contentRect,
-    imgixParams = {},
-    onLoad,
-    disableLibraryParam,
-    src,
-    children,
-    className = "",
-  } = props;
-  const { w: forcedWidth, h: forcedHeight } = imgixParams;
-  const hasDOMDimensions =
-    contentRect.bounds.width != null && contentRect.bounds.height != null;
-  const htmlAttributes = props.htmlAttributes || {};
-  const dpr = toFixed(2, imgixParams.dpr || global.devicePixelRatio || 1);
-  const ref = htmlAttributes.ref;
-  const onRef = (el) => {
-    measureRef(el);
-    if (typeof ref === "function") {
-      ref(el);
-    }
-  };
+class BackgroundImpl extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+
+  shouldComponentUpdate(nextProps) {
```

Comment:
This also needs to handle cases when other props change, e.g. `imgixParams`, `disableLibraryParam`, etc.